fabrics.c

```diff
@@ -1379,12 +1397,8 @@ static int do_discover(char *argstr, bool connect, enum nvme_print_flags flags)
 	if (instance < 0)
 		return instance;
 
-	if (asprintf(&dev_name, "/dev/nvme%d", instance) < 0)
```

Comment:
Something wrong with asprintf()?

Reply:
Nothing wrong with asprintf(), but since we're printing string that has a fixed and known max length (i.e. we're printing an int with %d, and MAX_INT=2147483647, which requires 11 char + 1 nul), we can simply allocate a buffer from the stack instead of the heap. This also eliminate the need to free() the memory. I understand that nvme-cli typically runs quickly to completion so we need not worry about memory leaks. However, there are cases where we can daemonize nvme-cli (e.g. nvme-monitor). I think it's just good practice to avoid malloc'ating memory when all we need is a small temporary buffer.